[FLINK-25129][docs] Improvements to the table-planner-loader related docs (#18812)
apache/flink: docs/content/docs/dev/table/sourcesSinks.md

If you want to implement a custom connector or a custom format, the following dependency is usually 
sufficient:

{{< artifact_tabs flink-table-common withProvidedScope >}}

If you want to develop a connector that needs to bridge with DataStream APIs (i.e. if you want to adapt
a DataStream connector to the Table API), you need to add this dependency:

{{< artifact_tabs flink-table-api-java-bridge withProvidedScope >}}

> When developing the connector/format, we suggest shipping both a thin JAR and an uber JAR. 

Reviewer: Can we add a quick sentence to explain why?

> In particular, the uber JAR should include all the 3rd party dependencies of the connector, 
> excluding the table dependencies listed above.

Reviewer: ```suggestion
In particular, the uber JAR should include all the third-party dependencies of the connector, 
excluding the table dependencies listed above.
```

This way, users can easily load the uber JAR in the SQL client or in the Flink distribution and start using it.

{{< hint warning >}}

> You MUST NOT depend on `flink-table-planner{{< scala_version >}}` in production code, 
> as with the new module `flink-table-planner-loader` introduced in Flink 1.15, 
> application's classpath won't have direct access to `org.apache.flink.table.planner` classes anymore. 

Reviewer: ```suggestion
You should not depend on `flink-table-planner{{< scala_version >}}` in production code. 
With the new module `flink-table-planner-loader` introduced in Flink 1.15, the
application's classpath will not have direct access to `org.apache.flink.table.planner` classes anymore. 
```

{{< artifact flink-table-api-java-bridge withProvidedScope >}}

When developing the connector/format, we suggest shipping both a thin JAR and an uber JAR. 

> In particular, the uber JAR should include all the 3rd party dependencies of the connector, excluding the table dependencies listed above.

Author: WDYM by line breaks? just one break for improving the markdown source, or two line breaks to add the empty line between text blocks?

> In particular, the uber JAR should include all the 3rd party dependencies of the connector, excluding the table dependencies listed above.

Reviewer 2: Sorry, I meant to avoid long lines.
